### PR TITLE
Handle multiple data files in Monte Carlo runs

### PR DESCRIPTION
**Pull request by rplzzz**
When we scale up our Monte Carlo runs, we end up splitting results into multiple output files.  Previously the filenames of those files were not being recorded correctly.  Furthermore, the analysis functions previously assumed that the results were all in a single data file, making it impossible to analyze the results produced by the scaled-up runs.  This change fixes both of these problems.

### CODE REVIEW

**Comment by rplzzz** [comment]
@kvcalvin Any objection to merging this?

File: R/bayesian.R [hunk 1]
```diff
@@ -575,3 +603,86 @@ hpdi_single <- function(samples, interval, reportvars, weighted, lp)
...
+#' calculations to scenario outputs where they were never calculated in the
+#' first place.  Either way, it only makes sense to do this with uniformly
+#' sampled, posterior-weighted samples.  If you do this with Markov Chain Monte
+#' Carlo samples, the recorded likelihood and posterior probability values will
+#' change, but the sampling will still reflect the old probability density.
+#'
+#' tldr: \strong{Do not use this function on the results of a Markov Chain Monte
+#' Carlo analysis!}  If you do, not only will you get bogus results, you will
+#' bring shame and disgrace on yourself and everyone you have ever worked with.
+#' Don't do it.
```
Comment (kvcalvin) [inline R/bayesian.R]: well, this is a little direct

File: R/bayesian.R [hunk 2]
```diff
@@ -575,3 +603,86 @@ hpdi_single <- function(samples, interval, reportvars, weighted, lp)
...
+    indx_stop <- indx_strt + 5
+
+    if(is.na(saveopt)) {
+        outdir <- file.path(inputdir, 'revised')
+        if(!dir.exists(outdir)) {
+            dir.create(outdir)
+        }
+    }
+
+    foreach(fn = sifiles) %dopar% {
```
Comment (kvcalvin) [inline R/bayesian.R]: maybe a few more comments in this block?

File: R/bayesian.R (hunk 1 above)
Comment (rplzzz) [reply R/bayesian.R]: It's maybe a little over the top.  I'm not sure whether this function really belongs in the final product or not.  I originally developed it to go back and do the probability density calculations on a bunch of runs where we had initially done just the model runs.  It's not clear that it has a use going forward, but it's also not clear that we _won't_ want to use it again.  Hence the dire warnings about not using it unless you're sure it's appropriate.

I can do some cleanup on this function (and its docs), or we can just remove it entirely.  I guess a middle way would be to demote it to a script and include it in `inst/scripts`.  What do you think, @kvcalvin?  Keep it or cut it?

Comment (kvcalvin) [reply R/bayesian.R]: Let's move it to scripts if it isn't critical for the package

Comment (rplzzz) [reply R/bayesian.R]: Ok, done.

**Comment by rplzzz** [comment]
@kvcalvin Any objection to merging this one?

**Comment by kvcalvin** [comment]
Nope. Although, we should figure out what is going on with codecov.

**Comment by rplzzz** [comment]
The codecov thing is a non issue.  Only 65% of the lines touched in this patch have test coverage because a lot of the code we updated in this one is deliberately not automatically tested (because functions like `run_ensemble` take longer than travis is willing to wait).  Since codecov's default settings have a target of 80%, it reports a failure, but it's an expected failure.

I'll edit the exclusion list in the codecov command to better reflect what we intend to test.  Right now `run_ensemble` is excluded, but it looks like some functions called only from `run_ensemble` are not.

**Comment by rplzzz** [comment]
Success.  Huzzah!

**Pull request merged**